kore/src/Kore/Reachability/Prove.hs

```diff
@@ -497,12 +537,19 @@ throwStuckClaims ::
         (VerifierT m)
         rule
         (ProofDepth, ClaimState SomeClaim)
-throwStuckClaims rule prim state = do
+throwStuckClaims maxCounterexamples rule prim state = do
```

Comment:
Sorry, I can't remember, did you end up finding out the reason why we need both `throwStuckClaims` and `throwUnproven`? Also, does this need `checkLeftUnproven` as well?

Reply:
From what I understand, first we had `throwUnproven`, but then `throwStuckClaims` was added in order to stop the execution earlier in the case of a stuck claim. Otherwise, during a breadth-first proof search, the backend would have run all the branches until proven or stuck (see https://github.com/kframework/kore/issues/2071).
I don't think `throwStuckClaims` needs `checkLeftUnproven` because `throwUnproven` is the last check.